core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -1888,7 +1888,13 @@ public void receivePending(Transaction tx, @Nullable List<Transaction> dependenc
                 return;
             if (isTransactionRisky(tx, dependencies) && !acceptRiskyTransactions) {
                 // isTransactionRisky already logged the reason.
-                riskDropped.put(tx.getTxId(), tx);
+
+                // Clone transaction to avoid multiple wallets pointing to the same transaction. This can happen when
+                // two wallets depend on the same transaction.
+                Transaction cloneTx = tx.getParams().getDefaultSerializer().makeTransaction(tx.bitcoinSerialize());
+                cloneTx.setPurpose(tx.getPurpose());
```

Comment:
Maybe `updatedAt` should be cloned, too?

Just for the record, I see you're not cloning the `exchangeRate` and `memo` fields. This is probably ok, since they are kind of wallet specific (though they should be in the WalletTransaction class if that's true).

Reply:
Agreed, `updatedAt` makes sense to clone as well, fixed in https://github.com/bitcoinj/bitcoinj/pull/2150/commits/804612ed35856ae04fd02718a1466959a78331df

I think it doesn't make that much sense to clone `exchangeRate` and `memo` since the cloning is done on transactions received from the network. Probably won't hurt either, and it might make sense for Transaction to have a `clone()` method that would handle that. Probably for another PR as you say.